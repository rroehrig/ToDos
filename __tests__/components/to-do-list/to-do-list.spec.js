import React from 'react';
import { shallow } from 'enzyme';
import ToDoList from '../../../app/components/to-do-list/to-do-list';

describe('ToDoList', () => {
    const mockedOnToDoPress = jest.fn();

    const firstToDo = {
        text: 'my first Todo',
        id: 1,
        completed: false,
    };
    const toDos = [firstToDo];
    const toDoWrapper = shallow(<ToDoList onToDoPress={mockedOnToDoPress} toDos={toDos} />);
    // Snapshot testing
    it('renders a ToDoList with one task correctly in a snapshot', () => {
        expect(toDoWrapper).toMatchSnapshot();
    });
});
