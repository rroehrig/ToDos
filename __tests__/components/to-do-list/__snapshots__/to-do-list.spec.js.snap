// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList renders a ToDoList with one task correctly in a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ListViewMock
    dataSource={
        ListViewDataSource {
            "items": 1,
          }
    }
    renderRow={[Function]}
    renderScrollComponent={[Function]}
/>,
  "nodes": Array [
    <ListViewMock
      dataSource={
            ListViewDataSource {
                  "items": 1,
                }
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ToDoList
        onToDoPress={[Function]}
        toDos={
                Array [
                        Object {
                          "completed": false,
                          "id": 1,
                          "text": "my first Todo",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onToDoPress": [Function],
          "toDos": Array [
            Object {
              "completed": false,
              "id": 1,
              "text": "my first Todo",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ListViewMock
          dataSource={
                    ListViewDataSource {
                              "items": 1,
                            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <ListViewMock
          dataSource={
                    ListViewDataSource {
                              "items": 1,
                            }
          }
          renderRow={[Function]}
          renderScrollComponent={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ToDoList
    onToDoPress={[Function]}
    toDos={
        Array [
            Object {
              "completed": false,
              "id": 1,
              "text": "my first Todo",
            },
          ]
    }
/>,
}
`;
